### PR TITLE
Extract interface ISerialPortStream for better testing/mocking

### PR DESCRIPTION
**Pull request by tedvanderveen**
For UnitTests, there are no actual Com ports and connected devices available, so in an ideal world creating a Mock (using Moq for example) requires no actual implementation, just an interface to mock.

### CODE REVIEW

File: code/ISerialPortStream.cs [hunk 1]
```diff
@@ -0,0 +1,771 @@
...
+        /// Returns a <see cref="System.String" /> that represents this instance.
+        /// </summary>
+        /// <returns>
+        /// A <see cref="System.String" /> that represents this instance.
+        /// </returns>
+        string ToString();
+
+        void CopyTo(Stream destination);
+        void CopyTo(Stream destination, int bufferSize);
+        Task CopyToAsync(Stream destination);
```
Comment (jcurl) [inline code/ISerialPortStream.cs]: The *Async won't compile for .NET 4.0, which I still use

File: code/ISerialPortStream.cs [hunk 2]
```diff
@@ -0,0 +1,771 @@
...
+
+        /// <summary>
+        /// Closes the port connection, sets the IsOpen property to false. Does not dispose the object.
+        /// </summary>
+        /// <remarks>
+        /// This method will clean up the object so far as to close the port. Internal buffers remain
+        /// active that the stream can continue to read. Writes will throw an exception.
+        /// </remarks>
+#if !NETSTANDARD15
+        public new void Close()
```
Comment (jcurl) [inline code/ISerialPortStream.cs]: This doesn't compile for NETFX.

**Comment by jcurl** [comment]
This is now in v2.x branch after cleaning up.

**Comment by tedvanderveen** [comment]
@jcurl can you please publish new Nuget package with this new ISerialPortStream reflected?

**Pull request closed without merging**